inference-engine/src/gna_plugin/frontend/scale_factor_calc.hpp

```diff
@@ -1337,6 +1338,39 @@ class ScaleFactorCalculator {
         needRestart = true;
         return true;
     }
-};
+
+    template<class T>
+    static int GetMandatoryWeightsBytesSize(T ptr) {
+        auto info = LayerInfo(ptr);
+        if (info.isConvolution() || info.isScaleShift()) {
```

Comment:
Do you have an understanding of why we use optional types for these operations? What does optional type mean?

Reply:
As far as I understood, depending on config.gnaPrecision model is quantized using 8 or 16 bits. Some layers, such as convolution, use 16-bit weights. For this reason, an optional type (16 bits) is used for convolution.

Reply:
@dmitriikhurtin As for me, "optional" is a very strange definition for this case. Convolution doesn't support I8 in GNA 1/2. So I16 is a mandatory type of weights for a specific operation on specific target. 

Related to ScaleShift it is not HW limitation. I found out notes about "Some layers will be switched into I16 mode  to not lose accuracy while memory and runtime performance of layers like scaleshifts still OK since it is O(N) ." 

I think it would be better to refactor this code to have direct mapping with appropriate comments what and why is used for specific operations. One "optional" type for all possible limitations looks strange.

Reply:
added comments